### PR TITLE
types: replace RoundMode by int32 type

### PR DESCRIPTION
**Pull request by erjiaqing**
### What problem does this PR solve? 

Replace RoundMode by int32 type to increase performance slightly.

### What is changed and how it works?

Replace RoundMode by int32 type, and set ModeHalfEven, ModeTruncate and modeCeiling to 5, 10 and 0. (Orig: 880ns / call -> 840ns / call)

I also found that we use too many divide constant and % constant, so I also make following changes:
1. build mod9 table, (840ns / call -> 820ns / call)
2. build div9 table, (820ns / call -> 783ns / call)

Thus we run 11% faster on this function.

### Check List 

Tests 

 - Passed existing tests.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: types/mydecimal.go [hunk 1]
```diff
@@ -49,18 +49,52 @@ const (
...
+		0, 1, 2, 3, 4, 5, 6, 7, 8,
+		0, 1, 2, 3, 4, 5, 6, 7, 8,
+		0, 1, 2, 3, 4, 5, 6, 7, 8,
+		0, 1, 2, 3, 4, 5, 6, 7, 8,
+		0, 1, 2, 3, 4, 5, 6, 7, 8,
+		0, 1, 2, 3, 4, 5, 6, 7, 8,
+		0, 1, 2, 3, 4, 5, 6, 7, 8,
+		0, 1,
+	}
+	div9 = [128]int{
```
Comment (winoros) [inline types/mydecimal.go]: Its format is different with the about `mod9`.

Comment (erjiaqing) [reply types/mydecimal.go]: `go fmt` did that

File: types/mydecimal.go [hunk 2]
```diff
@@ -174,6 +208,9 @@ func countTrailingZeroes(i int, word int32) int {
 }
 
 func digitsToWords(digits int) int {
+	if digits > 0 && digits < 118 {
```
Comment (ngaut) [inline types/mydecimal.go]: Could you add comments for magic number 118?

**Comment by breeswish** [comment]
In TiKV round mode is implemented as an enum, which internally will be turned into integer types, so no need to change in TiKV.

File: types/mydecimal.go (hunk 1 above)
Comment (winoros) [reply types/mydecimal.go]: Oh i know the reason.

**Pull request merged**